src/plugin.js

```diff
@@ -25,52 +28,67 @@ export class Title extends Component {
     this.title.appendChild(document.createTextNode(title));
     this.description.appendChild(document.createTextNode(description));
   }
-};
+}
 
+/**
+ * Shelf Component
+ */
 export class Shelf extends Component {
   createEl() {
     return super.createEl('div', {
       className: 'vjs-dock-shelf'
     });
   }
-};
+}
 
 videojs.registerComponent('Title', Title);
 videojs.registerComponent('Shelf', Shelf);
 
-videojs.plugin('dock', function(options) {
-  let player = this;
+/**
+ * A video.js plugin.
+ *
+ * In the plugin function, the value of `this` is a video.js `Player`
+ * instance. You cannot rely on the player being in a "ready" state here,
+ * depending on how the plugin is invoked. This may or may not be important
+ * to you; if not, remove the wait for "ready"!
+ *
+ * @function dock
+ * @param    {Object} [options={}]
+ *           An object of options left to the plugin author to define.
+ */
+const dock = function(options) {
   let opts = options || {};
-  let settings = assign({
-    title: {
-      title: '',
-      description: ''
-    }
-  }, {
+  let settings = {
```

Comment:
@gkatsev Per our discussion I started to go back to mergeOptions but it was giving me errors due to the code trying to write to the read-only options later. Going to just leave this as is for the purposes of this PR, definitely still room for improvement though.